python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py

```diff
@@ -571,6 +602,12 @@ def build(cls, workspace: WorkspaceSnapshot):
                 asset_infos_by_key[asset_node.key].append(
                     RepositoryScopedAssetInfo(
                         asset_node=asset_node,
+                        targeting_sensor_names=[
+                            s.name for s in repo.get_sensors_targeting(asset_node.key)
```

Comment:
If I'm understanding correctly, it looks like we need to make this call once per asset, and this call iterates through each sensor in the repo.

So this is O(#assets * #sensors), which could definitely add up. Also, the asset selection resolution logic could technically be O(#assets) as well (e.g. `AssetSelection.groups()` iterates through every asset), so we get to the kinda gnarly O(#assets^2 * #sensors) in the worst case.

Seems like we could do a first pass iteration which goes through each sensor in the repo, creates a mapping from job name to sensors, as well as a mapping of resolved asset key to sensors. Then when resolving this for a single asset, we just union sensors for its jobs and the sensors for its key, which works out to O(#sensors + #assets)

Reply:
[1] was my initial response to this idea as I was working on this, but I think you are right building up some mappings to index in to is the probably the way to go 

I should probably do some perf measures for before and after this stack to be safe

Reply:
maps made